pandas-stubs/core/groupby/generic.pyi

```diff
@@ -72,14 +76,24 @@ class SeriesGroupBy(GroupBy[Series[S1]], Generic[S1, ByT]):
         **kwargs,
     ) -> Series: ...
     agg = aggregate
+    @overload
     def transform(
         self,
-        func: Callable | str,
-        *args,
+        func: Callable[Concatenate[Series[S1], P], Series[S2]],
+        *args: Any,
```

Comment:
Can only define it as `Any` here, can't do `P.args` since engine argument gets in the middle.

Reply:
Could work if we rearranged the order, not sure if it is doable, open to suggestion.